Examples/Detectors/MagneticField/src/MagneticFieldOptions.cpp

```diff
@@ -0,0 +1,180 @@
+      "octant/quadrant and the field is symmetrically extended to the full "
+      "space.");
+  opt("bf-map-lengthscale-mm", value<double>()->default_value(1.),
+      "Optional length scale modifier for the field map grid. This options "
+      "only needs to be set if the length unit in the field map file is not "
+      "`mm`. The value must scale from the stored unit to the equivalent value "
+      "in `mm`.");
+  opt("bf-map-fieldscale-tesla", value<double>()->default_value(1.),
+      "Optional field value scale modifier for the field map value. This "
+      "option only needs to be set if the field value unit in the field map "
+      "file is not `Tesla`. The value must scale from the stored unit to the "
+      "equvalent value in `Tesla`.");
+}
+
+void ActsExamples::Options::setupMagneticFieldServices(const Variables& vars,
+                                                       Sequencer& sequencer) {
+  if (vars["bf-scalable"].as<bool>()) {
+    ScalableBFieldService::Config sbfCfg;
+    sbfCfg.scalor = vars["bf-scalable-scalor"].as<double>();
+    sequencer.addContextDecorator(
+        std::make_shared<ScalableBFieldService>(sbfCfg, Acts::Logging::INFO));
+  }
+}
+
+ActsExamples::MagneticField ActsExamples::Options::readMagneticField(
+    const Variables& vars) {
+  using namespace ActsExamples::detail;
+  using boost::filesystem::path;
+
+  ACTS_LOCAL_LOGGER(
+      Acts::getDefaultLogger("MagneticField", Acts::Logging::INFO));
+
+  // first option: create a constant field
+  if (vars.count("bf-constant-tesla")) {
+    const auto values = vars["bf-constant-tesla"].as<Reals<3>>();
+    Acts::Vector3 field(values[0] * Acts::UnitConstants::T,
+                        values[1] * Acts::UnitConstants::T,
+                        values[2] * Acts::UnitConstants::T);
+    if (vars["bf-scalable"].as<bool>()) {
+      ACTS_INFO("Use a constant magnetic field with per-event scaling");
+      return std::make_shared<ScalableBField>(field);
+    } else {
+      ACTS_INFO("Use a constant magnetic field");
+      return std::make_shared<Acts::ConstantBField>(field);
+    }
+  }
+
+  // second option: read a field map from a file
+  if (vars.count("bf-map-file")) {
+    const path file = vars["bf-map-file"].as<std::string>();
+    const auto tree = vars["bf-map-tree"].as<std::string>();
+    const auto type = vars["bf-map-type"].as<std::string>();
+    const auto useOctantOnly = vars["bf-map-octantonly"].as<bool>();
+    const auto lengthUnit =
+        vars["bf-map-lengthscale-mm"].as<double>() * Acts::UnitConstants::mm;
+    const auto fieldUnit =
+        vars["bf-map-fieldscale-tesla"].as<double>() * Acts::UnitConstants::T;
+
+    bool readRoot = false;
+    if (file.extension() == ".root") {
+      ACTS_INFO("Read magnetic field map from ROOT file '" << file << "'");
+      readRoot = true;
+    } else if (file.extension() == ".txt") {
+      ACTS_INFO("Read magnetic field map from text file '" << file << "'");
+      readRoot = false;
```

Comment:
It looks like this line can be omitted.

Reply:
Yes, it could be omitted. It is here intentionally so both parts of the `if/else` statement have the same structure and it is easier to understand.